File: writer-python/app/schemas/copilot.py
"""通用写作 Copilot 对话（场景化系统提示）。"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel, Field


class CopilotChatMessage(BaseModel):
    role: Literal["user", "assistant"]
    content: str = Field(max_length=32000)


class CopilotChatRequest(BaseModel):
    model_config = {"populate_by_name": True}

    project_id: str = Field(alias="projectId", min_length=1)
    scene: Literal["init_wizard", "outline_edit", "chapter_coach"]
    messages: list[CopilotChatMessage] = Field(min_length=1)
    context_blob: str | None = Field(default=None, alias="contextBlob", max_length=48000)


class CopilotChatResponse(BaseModel):
    reply: str

File: frontend/src/api/copilot.ts
import { apiJson } from "./client";

export type CopilotScene = "init_wizard" | "outline_edit" | "chapter_coach" | "setup_coach";

export type CopilotChatMessage = { role: "user" | "assistant"; content: string };

export async function postCopilotChat(body: {
  projectId: string;
  scene: CopilotScene;
  messages: CopilotChatMessage[];
  contextBlob?: string;
}): Promise<{ reply: string }> {
  return apiJson<{ reply: string }>("/api/writer/copilot/chat", {
    method: "POST",
    body: JSON.stringify({
      projectId: body.projectId,
      scene: body.scene,
      messages: body.messages,
      contextBlob: body.contextBlob,
    }),
  });
}
